### ISSUE TITLE
zcnsc panic if no authorisers

### ISSUE THREAD

**Issue by Sriep** [0chain/0chain#2026]
If we do a mint transaction when there are no authorisers, then we could get a panic.

If [getAuthorizerCount returns zero then the signatures slice is emptied.](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/zcnsc/mint.go#L50)
```go
	if len(payload.Signatures) > numAuth {
		logging.Logger.Info("no of signatures execeed the no of available authorizers", zap.Int("available", numAuth))
		payload.Signatures = payload.Signatures[0:numAuth]
	}
```
We then[ divide by the number of signatures](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/zcnsc/mint.go#L121)
```go
share, _, err = currency.DistributeCoin(gn.ZCNSConfig.MaxFee, int64(len(payload.Signatures)))
```
We should return an error in this case rather than panic.

Also we do not[ check the error ](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/zcnsc/mint.go#L50)when we get the authoriser count
```go
numAuth, err := getAuthorizerCount(ctx)
```
handling this error will probably stop the panic as well.

Note that the [zcnsc.mint](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/zcnsc/benchmark_tests.go#L92) benchmark test is failing due to a panic rather than a returned error because of this issue.

**Comment by peterlimg** [comment]
@Sriep the panic is normal as we do zero division, what we need to do is check the authorizer number and return error if it's empty.

**Comment by peterlimg** [comment]
Fixed via #2036
